src/Core/Misc/ExtensionMethods.cs

```diff
@@ -692,5 +693,10 @@ public static T GetPropertyOrDefault<T>(this A11yElement e, int propertyId)
 
             return a != null;
         }
+
+        public static string WithParameters(this string formatString, params object[] args)
+        {
+            return string.Format(CultureInfo.InvariantCulture, formatString, args);
```

Comment:
We are likely to have some formats that will need InvariantCulture, while others will need CurrentCulture. That suggests 2 extension methods, plus an audit to select the appropriate method. Is this in scope for this PR or is that a future task?

Reply:
I'd prefer to make that a separate PR, as this one is already so large. What do you think the determining factor is for if it needs to be invariant or not?

Reply:
Makes sense. As I understand it, our current bar is that it needs to be CurrentCulture if it's in the data that comes back from the `IScanner` interface. Messages for `AxeWindowsAutomationException` should presumably be localized and use CurrentCulture, since they're for consuming developers. Other Exception messages are for us and are probably out of scope for localization or CurrentCulture.

Reply:
Gotcha. I made the change in this PR by searching/replacing CurrentCulture.InvariantCulture, so this shouldn't touch any of the CurrentCulture instances at the moment and I think it makes sense to address that in a separate PR. Maybe it makes sense to have different extension methods for the different scenarios:

- An extension for exceptions, that will always use InvariantCulture
- An extension for AxeWindowsAutomatedExceptions, which may use either Current or Invariant culture depending on if we end up deciding to localize
- And an extension for message exposed to the user

What do you think?